consensus/hotstuff/signature/consensus_test.go

```diff
@@ -0,0 +1,49 @@
+package signature
+
+import (
+	"testing"
+
+	"github.com/onflow/flow-go/consensus/hotstuff"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/stretchr/testify/require"
+)
+
+func TestEncodeDecodeStakingSig(t *testing.T) {
+	sig := unittest.SignatureFixture()
+	encoded := EncodeSingleSig(hotstuff.SigTypeStaking, sig)
+	decodedType, decodedSig, err := DecodeSingleSig(encoded)
+	require.NoError(t, err)
+	require.Equal(t, hotstuff.SigTypeStaking, decodedType)
+	require.Equal(t, sig, decodedSig)
+}
+
+func TestEncodeDecodeRandomBeaconSig(t *testing.T) {
+	sig := unittest.SignatureFixture()
+	encoded := EncodeSingleSig(hotstuff.SigTypeRandomBeacon, sig)
+	decodedType, decodedSig, err := DecodeSingleSig(encoded)
+	require.NoError(t, err)
+	require.Equal(t, hotstuff.SigTypeRandomBeacon, decodedType)
+	require.Equal(t, sig, decodedSig)
+}
+
+func TestDecodeEmptySig(t *testing.T) {
+	_, _, err := DecodeSingleSig([]byte{})
+	require.Error(t, err)
+}
+
+// encode two different sigs wth the same type, the encoded sig should be different
+func TestEncodeTwoSigsDifferent(t *testing.T) {
+	sigs := unittest.SignaturesFixture(2)
+	sig1, sig2 := sigs[0], sigs[1]
+	encodedSig1 := EncodeSingleSig(hotstuff.SigTypeStaking, sig1)
+	encodedSig2 := EncodeSingleSig(hotstuff.SigTypeStaking, sig2)
+	require.NotEqual(t, encodedSig1, encodedSig2)
+}
+
+// encode the same sig with the different type, the encoded sig should be different
+func TestEncodeSameSigWithDifferentTypeShouldBeDifferen(t *testing.T) {
+	sig := unittest.SignatureFixture()
+	encodedAsStaking := EncodeSingleSig(hotstuff.SigTypeStaking, sig)
+	encodedAsRandomBeacon := EncodeSingleSig(hotstuff.SigTypeRandomBeacon, sig)
+	require.NotEqual(t, encodedAsStaking, encodedAsRandomBeacon)
+}
```

Comment:
The test `TestEncodeDecodeSig` formally includes these 2 tests: it means that these 2 tests are already tested by `TestEncodeDecodeSig`, and in theory are not needed.